### DISCUSSION TITLE
What is the correct way to get a Pimore object field definitions class by its shortname?

### DISCUSSION THREAD

**Discussion by Blackbam** [pimcore/pimcore#9216] [Q&A / Support]
First of all I tried to get the mapping like this:

```php bin/console debug:container --parameters```

Problem is that the config values seem not to be available in there.

After searching a little bit I found the shortname to field definitions mapping in the Pimcore CoreBundle default.yml file. Therefore this is what I am doing in order to get this mapping:

```
    /**
     * @return array: An array of the pimcore default class definitions 
     */
    public function getFieldDefinitionMapping(): array {
        $pimcoreDefaultConfig = Yaml::parse(
            file_get_contents(PIMCORE_PROJECT_ROOT.'/vendor/pimcore/pimcore/bundles/CoreBundle/Resources/config/pimcore/default.yml')
        );

        return $pimcoreDefaultConfig["pimcore"]["objects"]["class_definitions"]["data"]["map"];
    }
```

Still I guess there must be a better way - this e.g. will not work if some of the config values are overridden somewhere. Can somebody tell me how to get this correctly?

**Comment by NiklasBr** [comment]
Have you tried `\Pimcore\Model\DataObject\ClassDefinition::getByName(\Pimcore\Model\DataObject\Product::classId())`? (change `Product` to the class you want to use)

**Reply by Blackbam** [reply]
Thank you very much for your quick reply! What I am actually searching for is a mapping of a class definition data field to its short name like this from the config:

```
block: \Pimcore\Model\DataObject\ClassDefinition\Data\Block
calculatedValue: \Pimcore\Model\DataObject\ClassDefinition\Data\CalculatedValue
checkbox: \Pimcore\Model\DataObject\ClassDefinition\Data\Checkbox
classificationstore: \Pimcore\Model\DataObject\ClassDefinition\Data\Classificationstore
consent: \Pimcore\Model\DataObject\ClassDefinition\Data\Consent
country: \Pimcore\Model\DataObject\ClassDefinition\Data\Country
countrymultiselect: \Pimcore\Model\DataObject\ClassDefinition\Data\Countrymultiselect
date: \Pimcore\Model\DataObject\ClassDefinition\Data\Date
datetime: \Pimcore\Model\DataObject\ClassDefinition\Data\Datetime
email: \Pimcore\Model\DataObject\ClassDefinition\Data\Email
encryptedField: \Pimcore\Model\DataObject\ClassDefinition\Data\EncryptedField
....
```

I guess it can be somehow accessed via config?

**Comment by dvesh3** [comment]
@Blackbam you can use the implementation loader like:
```php
$loader = \Pimcore::getContainer()->get('pimcore.implementation_loader.object.data');
$blockClassName = $loader->getClassNameFor('block');
```
